test/src/tests/widgets/test_button.py

```diff
@@ -0,0 +1,19 @@
+from unittest.mock import Mock
+
+from pytest import fixture
+
+import toga
+
+
+@fixture
+async def new_widget():
```

Comment:
> Do all fixtures have to be async?

Yes, if they call any Toga APIs, because making them async is what causes them to run on the main thread.

> Is a fixture the right model here - especially for the "broader layout"? Might it not be easier to have a `simple_layout(widget)` which puts the test widget into a simple box layout?

I'm not quite sure what you're suggesting, but I'll rename the fixtures so it's clearer what each one is actually doing.